src/CattyTests/DataModel/UserDataContainerTest.swift

```diff
@@ -452,36 +453,35 @@ final class UserDataContainerTest: XCTestCase {
         let userVariable1 = UserVariable(name: "testName1")
         let userVariable2 = UserVariable(name: "testName2")
         let userVariable3 = UserVariable(name: "testName3")
-        let userVariable4 = UserVariable(name: "testName4")
 
-        var result = container.addObjectVariable(userVariable1, for: objectA)
+        var result = objectA.userData.addVariable(userVariable1)
         XCTAssertTrue(result)
 
-        result = container.addObjectVariable(userVariable2, for: objectB)
+        result = objectB.userData.addVariable(userVariable2)
         XCTAssertTrue(result)
 
-        container.programVariableList.add(userVariable3)
+        container.addVariable(userVariable3)
 
         XCTAssertEqual(3, UserDataContainer.allVariables(for: project).count)
         XCTAssertEqual(2, UserDataContainer.objectAndProjectVariables(for: objectA).count)
         XCTAssertEqual(2, UserDataContainer.objectAndProjectVariables(for: objectB).count)
 
-        XCTAssertFalse(container.removeUserVariableNamed(userVariable1.name, for: objectB))
-        XCTAssertTrue(container.removeUserVariableNamed(userVariable1.name, for: objectA))
-        XCTAssertFalse(container.removeUserVariableNamed(userVariable1.name, for: objectA))
+        XCTAssertFalse(objectB.userData.removeUserVariable(withName: userVariable1.name))
+        XCTAssertTrue(objectA.userData.removeUserVariable(withName: userVariable1.name))
+        XCTAssertFalse(objectA.userData.removeUserVariable(withName: userVariable1.name))
 
         XCTAssertEqual(2, UserDataContainer.allVariables(for: project).count)
         XCTAssertEqual(1, UserDataContainer.objectAndProjectVariables(for: objectA).count)
         XCTAssertEqual(2, UserDataContainer.objectAndProjectVariables(for: objectB).count)
 
-        XCTAssertTrue(container.removeUserVariableNamed(userVariable3.name, for: objectB))
-        XCTAssertFalse(container.removeUserVariableNamed(userVariable3.name, for: objectB))
+        XCTAssertTrue(objectB.userData.removeUserVariable(withName: userVariable2.name))
```

Comment:
Instead of `userVariable2`, `userVariable3` should be removed from the container.